docs/doxygen/man/man3/Connected.3
.TH "Connected" 3 "Thu Aug 30 2018" "esc" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Connected \- \fBConnected\fP message class\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <connected\&.h>\fP
.PP
Inherits \fBBlockCommand\fP\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConnected\fP (uint16_t port, uint32_t ip_address, std::string version)"
.br
.ti -1c
.RI "virtual int \fBgetType\fP () override"
.br
.RI "\fIReturn TXSTYPE_CON as type \&. \fP"
.ti -1c
.RI "virtual CommandType \fBgetCommandType\fP () override"
.br
.RI "\fIReturn eReadingOnly as command type \&. \fP"
.ti -1c
.RI "virtual unsigned char * \fBgetData\fP () override"
.br
.RI "\fIGet pointer to command data structure\&. \fP"
.ti -1c
.RI "virtual unsigned char * \fBgetResponse\fP () override"
.br
.RI "\fIGet pointer to response data\&. \fP"
.ti -1c
.RI "virtual void \fBsetData\fP (char *data) override"
.br
.RI "\fIPut data as a char list and put convert it to data structure\&. \fP"
.ti -1c
.RI "virtual void \fBsetResponse\fP (char *response) override"
.br
.RI "\fIApply data as a response struct\&. \fP"
.ti -1c
.RI "virtual int \fBgetDataSize\fP () override"
.br
.RI "\fIGet data struct size\&. Without signature\&. \fP"
.ti -1c
.RI "virtual int \fBgetResponseSize\fP () override"
.br
.RI "\fIGet response data struct size\&. \fP"
.ti -1c
.RI "virtual unsigned char * \fBgetSignature\fP () override"
.br
.RI "\fIGet pointer to signature data\&. \fP"
.ti -1c
.RI "virtual int \fBgetSignatureSize\fP () override"
.br
.RI "\fIGet signature size\&. \fP"
.ti -1c
.RI "virtual void \fBsign\fP (const uint8_t *hash, const uint8_t *sk, const uint8_t *pk) override"
.br
.RI "\fISign actual data plus hash using user private and public keys\&. \fP"
.ti -1c
.RI "virtual bool \fBcheckSignature\fP (const uint8_t *hash, const uint8_t *pk) override"
.br
.RI "\fICheck actual signature\&. \fP"
.ti -1c
.RI "virtual uint32_t \fBgetTime\fP () override"
.br
.RI "\fIGet time of command\&. \fP"
.ti -1c
.RI "virtual uint32_t \fBgetUserId\fP () override"
.br
.RI "\fIGet User ID\&. \fP"
.ti -1c
.RI "virtual uint32_t \fBgetBankId\fP () override"
.br
.RI "\fIGet Node ID\&. \fP"
.ti -1c
.RI "virtual int64_t \fBgetFee\fP () override"
.br
.RI "\fIGet command fee\&. \fP"
.ti -1c
.RI "virtual int64_t \fBgetDeduct\fP () override"
.br
.RI "\fIGet change in cash balance after command\&. \fP"
.ti -1c
.RI "virtual uint32_t \fBgetUserMessageId\fP () override"
.br
.RI "\fIGet message id\&. \fP"
.ti -1c
.RI "virtual bool \fBsend\fP (\fBINetworkClient\fP &netClient) override"
.br
.RI "\fISend data to the server\&. \fP"
.ti -1c
.RI "virtual void \fBsaveResponse\fP (\fBsettings\fP &sts) override"
.br
.RI "\fISave command response to settings object\&. \fP"
.ti -1c
.RI "virtual std::string \fBtoString\fP (bool pretty) override"
.br
.ti -1c
.RI "virtual void \fBtoJson\fP (boost::property_tree::ptree &ptree) override"
.br
.ti -1c
.RI "virtual void \fBtxnToJson\fP (boost::property_tree::ptree &ptree) override"
.br
.ti -1c
.RI "virtual std::string \fBusageHelperToString\fP () override"
.br
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "\fBConnectedInfo\fP \fBm_data\fP"
.br
.ti -1c
.RI "\fBcommandresponse\fP \fBm_response\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
\fBConnected\fP message class\&. 
.PP
Definition at line 8 of file connected\&.h\&.
.SH "Member Function Documentation"
.PP 
.SS "bool Connected::checkSignature (const uint8_t * hash, const uint8_t * pk)\fC [override]\fP, \fC [virtual]\fP"

.PP
Check actual signature\&. 
.PP
\fBParameters:\fP
.RS 4
\fIhash\fP Previous hash operation\&. 
.br
\fIpk\fP Pointer to public key\&. 
.RE
.PP

.PP
Implements \fBICommand\fP\&.
.PP
Definition at line 63 of file connected\&.cpp\&.
.SS "bool Connected::send (\fBINetworkClient\fP & netClient)\fC [override]\fP, \fC [virtual]\fP"

.PP
Send data to the server\&. 
.PP
\fBParameters:\fP
.RS 4
\fInetClient\fP Netwrok client implementation of \fBINetworkClient\fP interface\&. 
.br
\fIpk\fP Pointer to public key\&. 
.RE
.PP

.PP
Implements \fBICommand\fP\&.
.PP
Definition at line 95 of file connected\&.cpp\&.
.SS "void Connected::sign (const uint8_t * hash, const uint8_t * sk, const uint8_t * pk)\fC [override]\fP, \fC [virtual]\fP"

.PP
Sign actual data plus hash using user private and public keys\&. 
.PP
\fBParameters:\fP
.RS 4
\fIhash\fP Previous hash operation\&. 
.br
\fIsk\fP Pointer to provate key\&. 
.br
\fIpk\fP Pointer to public key\&. 
.RE
.PP

.PP
Implements \fBICommand\fP\&.
.PP
Definition at line 59 of file connected\&.cpp\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for esc from the source code\&.


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.0.0] - 2018-08-24
### Changed
- Genesis file for the launch
- Update README

## [0.0.6] - 2018-08-10
### Added
- Profit sharing among TOP and VIP nodes

### Changed
- Genesis file for test launch

### Fixed
- Vote counting in get_block command
- Ability to send externally signed transactions

## [0.0.5] - 2018-08-01
### Added
- Support passing version by cmake command.
- Contributing guide
- "Dry run" test case

### Fixed
- Node stops after incorrect `send_many` request  - #134
- Fee checking
- Internal transactions numbers

## [0.0.4] - 2018-07-27
### Changed
- Update default parameters
- Creating a dev version with reference to the last tag

[Unreleased]: https://github.com/adshares/ads/compare/v1.0.0...HEAD

[0.0.5]: https://github.com/adshares/ads/compare/v0.0.6...v1.0.0
[0.0.5]: https://github.com/adshares/ads/compare/v0.0.5...v0.0.6
[0.0.5]: https://github.com/adshares/ads/compare/v0.0.4...v0.0.5
[0.0.4]: https://github.com/adshares/ads/compare/v0.0.3...v0.0.4


src/common/command/connected.cpp
#include "connected.h"

#include <sstream>

#include "ed25519/ed25519.h"
#include "abstraction/interfaces.h"
#include "helper/json.h"

Connected::Connected()
    : m_data{} {
    m_responseError = ErrorCodes::Code::eNone;
}

Connected::Connected(uint16_t port, uint32_t ip_address, std::string version)
    : m_data(port, ip_address, version.c_str()) {
    m_responseError = ErrorCodes::Code::eNone;
}

unsigned char* Connected::getData() {
    return reinterpret_cast<unsigned char*>(&m_data);
}

unsigned char* Connected::getResponse() {
    return reinterpret_cast<unsigned char*>(&m_response);
}

void Connected::setData(char* data) {
    m_data = *reinterpret_cast<decltype(m_data)*>(data);
}

void Connected::setResponse(char* response) {
    m_response = *reinterpret_cast<decltype(m_response)*>(response);
}

int Connected::getDataSize() {
    return sizeof(m_data);
}

int Connected::getResponseSize() {
    return 0;
}

unsigned char* Connected::getSignature() {
    return (unsigned char*)"";
}

int Connected::getSignatureSize() {
    return 0;
}

int Connected::getType() {
    return TXSTYPE_CON;
}

CommandType Connected::getCommandType() {
    return CommandType::eReadingOnly;
}

void Connected::sign(const uint8_t* /*hash*/, const uint8_t* /*sk*/, const uint8_t* /*pk*/) {
    // no signature
}

bool Connected::checkSignature(const uint8_t* /*hash*/, const uint8_t* /*pk*/) {
    // no signature
    return true;
}

void Connected::saveResponse(settings& /*sts*/) {
}

uint32_t Connected::getUserId() {
    return 0;
}

uint32_t Connected::getBankId() {
    return 0;
}

uint32_t Connected::getTime() {
    return 0;
}

int64_t Connected::getFee() {
    return 0;
}

int64_t Connected::getDeduct() {
    return 0;
}

uint32_t Connected::getUserMessageId() {
    return 0;
}

bool Connected::send(INetworkClient& /*netClient*/) {
    return true;
}

std::string Connected::toString(bool /*pretty*/) {
    return "";
}

void Connected::toJson(boost::property_tree::ptree& /*ptree*/) {
}

void Connected::txnToJson(boost::property_tree::ptree& ptree) {
    std::stringstream ip_address{};
    ip_address << (m_data.ip_address & 0xFF) << "."
      << ((m_data.ip_address >> 8) & 0xFF) << "."
      << ((m_data.ip_address >> 16) & 0xFF) << "." << (m_data.ip_address >> 24);

    char verStr[17];
    uint i=0;
    while (i < sizeof(verStr)-1 && isprint(m_data.version[i]))
    {
        verStr[i] = m_data.version[i];
        i++;
    }
    verStr[i] = '\0';
    using namespace Helper;
    ptree.put(TAG::TYPE, getTxnName(m_data.ttype));
    ptree.put(TAG::PORT, m_data.port);
    ptree.put(TAG::IP_ADDRESS, ip_address.str());
    ptree.put(TAG::NODE_VERSION, verStr);
}

std::string Connected::usageHelperToString() {
    return std::string("");
}
